GitUI/UserControls/RevisionGrid/RevisionGridControl.cs

```diff
@@ -914,16 +913,46 @@ public void PerformRefreshRevisions(Func<RefsFilter, IReadOnlyList<IGitRef>> get
                 _gridView.SelectionChanged += OnGridViewSelectionChanged;
                 _gridView.ResumeLayout();
 
+                // Add back and show the spinner control, which was removed by SetPage call
+                Controls.Add(_loadingControlSync);
+                ShowLoading();
+
+                cancellationToken.ThrowIfCancellationRequested();
+
+                // Find all ambiguous refs (including stash, notes etc)
+                // Note: GetRefs can be very slow operation for large repos, start in parallel to git-log.
+                // refsByObjectId is needed when first output from git-log is handled
+                Func<IReadOnlyList<IGitRef>> refs = () => (getRefs ?? Module.GetRefs)(RefsFilter.NoFilter);
+                _ambiguousRefs = new(() => GitRef.GetAmbiguousRefNames(refs()));
+                refsByObjectId = new(() => refs().ToLookup(head => head.ObjectId));
```

Comment:
```suggestion
                refsByObjectId = new(() => refs().ToLookup(head => head.ObjectId), LazyThreadSafetyMode.ExecutionAndPublication);
```

This seems to be unnecessary for `_ambiguousRefs` because it looked as it was used from the UI thread only.

Reply:
The same Lazy handling is used for GetRefs()
https://docs.microsoft.com/en-us/dotnet/framework/performance/lazy-initialization#thread-safe-initialization
I interpret that as LazyThreadSafetyMode.ExecutionAndPublication is not needed, but the rest of the doc is unclear...

Changing this though (not for getRefs())